migration/v2.20.0-ck8sx-v2.21.0-ck8s1/upgrade-cluster.md

```diff
@@ -0,0 +1,130 @@
+
+- [ ] Notify the users (if any) before the upgrade starts;
+- [ ] Check if there are any pending changes to the environment;
+- [ ] Check the state of the environment, pods, nodes and backup jobs:
+
+    ```bash
+    ./compliantkubernetes-apps/bin/ck8s test sc|wc
+    ./compliantkubernetes-apps/bin/ck8s ops kubectl sc|wc get pods -A -o custom-columns=NAMESPACE:metadata.namespace,POD:metadata.name,READY-false:status.containerStatuses[*].ready,REASON:status.containerStatuses[*].state.terminated.reason | grep false | grep -v Completed
+    ./compliantkubernetes-apps/bin/ck8s ops kubectl sc|wc get nodes
+    ./compliantkubernetes-apps/bin/ck8s ops kubectl sc|wc get jobs -A
+    velero get backup
+    ```
+
+- [ ] Silence the notifications for the alerts. e.g you can use [alertmanager silences](https://prometheus.io/docs/alerting/latest/alertmanager/#silences);
+
+## Steps that can be done before the upgrade - non-disruptive
+
+1. Checkout the new release: `git switch -d v2.21.0-ck8s1`
+
+1. Switch to the correct remote: `git submodule sync`
+
+1. Update the kubespray submodule: `git submodule update --init --recursive`
+
+1. Add the following snippet at the end of both `sc-config/group_vars/k8s_cluster/ck8s-k8s-cluster.yaml` and `wc-config/group_vars/k8s_cluster/ck8s-k8s-cluster.yaml`
+
+    This enables NTP service and with multiple NTP servers, specificity in sweden.
+    You can visit [www.ntppool.org](https://www.ntppool.org/zone/@) to find other ntp pools if you are in other parts of the world.
+
+    ```yaml
+    ntp_enabled: true
+    ntp_manage_config: true
+    ntp_servers:
+    - "gbg1.ntp.netnod.se iburst"
+    - "gbg2.ntp.netnod.se iburst"
+    - "lul1.ntp.netnod.se iburst"
+    - "lul2.ntp.netnod.se iburst"
+    - "mmo1.ntp.netnod.se iburst"
+    - "mmo2.ntp.netnod.se iburst"
+    - "sth1.ntp.netnod.se iburst"
+    - "sth2.ntp.netnod.se iburst"
+    - "sth3.ntp.netnod.se iburst"
+    - "sth4.ntp.netnod.se iburst"
+    - "svl1.ntp.netnod.se iburst"
+    - "svl2.ntp.netnod.se iburst"
+    ntp_timezone: "Etc/UTC"
+    ```
+
+1. Disable Pod Security Policies
+
+    1. Change the following variable in both `sc-config/group_vars/k8s_cluster/ck8s-k8s-cluster.yaml` and `wc-config/group_vars/k8s_cluster/ck8s-k8s-cluster.yaml`
+
+        ```diff
+        -podsecuritypolicy_enabled: true
+        +podsecuritypolicy_enabled: false
+        ```
+
+    1. Remove `PodSecurityPolicy` from the `kube_apiserver_enable_admission_plugins` list in both `sc-config/group_vars/k8s_cluster/ck8s-k8s-cluster.yaml` and `wc-config/group_vars/k8s_cluster/ck8s-k8s-cluster.yaml`
+
+        ```diff
+         kube_apiserver_enable_admission_plugins:
+        -  - "PodSecurityPolicy"
+           - "NamespaceLifecycle"
+        ```
```

Comment:
I realized that I would also like to add
```suggestion
        ```

1. Set shutdown manager to valid values for Ubuntu

    ```bash
    kubelet_shutdown_grace_period: 30s
    kubelet_shutdown_grace_period_critical_pods: 10s
    ```
```